subsystems/jets/JetDraw.cc

```diff
@@ -0,0 +1,860 @@
+//Jennifer James <jennifer.l.james@vanderbilt.edu>, McKenna Sleeth,  and Derek Anderson
+#include "JetDraw.h"
+//#include "JetGoodRunChecker.h"
+#include <sPhenixStyle.C>
+#include <qahtml/QADrawClient.h>
+#include <qahtml/QADrawDB.h>
+#include <TCanvas.h>
+#include <TDatime.h>
+#include <TGraphErrors.h>
+#include <TH1.h>
+#include <TH2.h>
+#include <TLatex.h>
+#include <TString.h>
+#include <TPad.h>
+#include <TProfile.h>
+#include <TROOT.h>
+#include <TStyle.h>
+#include <TSystem.h>
+#include <TText.h>
+#include <TGaxis.h>
+#include <boost/format.hpp>
+#include <cmath>
+#include <ctime>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+
+JetDraw::JetDraw(const std::string &name)
+: cl(QADrawClient::instance())
+{
+  if (name == "JetQA") {
+    // Handle specific condition
+  }
+  memset(TC, 0, sizeof(TC));
+  memset(transparent, 0, sizeof(transparent));
+  memset(Pad, 0, sizeof(Pad));
+  DBVarInit();
+  histprefix = "h_constituentsinjets";
+  histprefix1 = "h_eventwiserho";
+  histprefix2 = "h_jetkinematiccheck";
+  histprefix3 = "h_jetseedcount";
+  return;
+}
+
+JetDraw::~JetDraw()
+{
+  /* delete db; */
+  return;
+}
+
+int JetDraw::MakeHtml(const std::string &what) {
+  if (what == "ALL") {
+    // Handle the "ALL" case here if necessary
+  } 
+  int iret = 0;
+  const int drawError = Draw(what);  // Call to Draw
+  if (drawError) {
+    return drawError;  // Return if there is an error in Draw
+  }
+    
+  int resolutionDrawError = 0;
+  for (uint32_t trigToDraw : m_vecTrigToDraw) {  // Loop over triggers
+    for (auto& resAndName : m_mapResToName) {  // Loop over resolution and name pairs
+      std::string pngFile = cl->htmlRegisterPage(
+						 *this,
+						 m_mapTrigToName[trigToDraw] + "/" + std::to_string(resAndName.first),
+						 m_mapTrigToName[trigToDraw] + "_" + resAndName.second,
+						 "png");
+      // Convert canvas to PNG
+      resolutionDrawError = cl->CanvasToPng(m_vecCanvas[trigToDraw][resAndName.first], pngFile);
```

Comment:
Correct me if I'm wrong, but I think as written nothing will actually be drawn on any of the canvases in `m_vecCanvas`? Rather, it looks like canvases get added to the vector, but everything actually gets drawn on the canvases in `TC`...

Reply:
Ah yes you're right. The actual drawing operations are not performed on the canvases stored in m_vecCanvas, but rather on separate canvases and then converted to PNG files using the cl->CanvasToPng() function. The purpose of the m_vecCanvas vector was to manage and store the canvases for each trigger and resolution, but the drawing happens via TC elsewhere